source/grid/grid_tools.cc

```diff
@@ -5882,67 +5887,104 @@ namespace GridTools
+            global_bboxes_to_be_used = &global_bboxes_temp;
+          }
+
+        // helper function to determine if a bounding box is valid
+        const auto is_valid = [](const auto &bb) {
+          for (unsigned int i = 0; i < spacedim; ++i)
+            if (bb.get_boundary_points().first[i] >
+                bb.get_boundary_points().second[i])
+              return false;
 
-          // linearize vector of vectors
-          std::vector<std::pair<BoundingBox<spacedim>, unsigned int>>
-            boxes_and_ranks;
+          return true;
+        };
 
-          for (unsigned rank = 0; rank < global_bboxes_to_be_used->size();
-               ++rank)
-            for (const auto &box : (*global_bboxes_to_be_used)[rank])
-              if (is_valid(box))
-                boxes_and_ranks.emplace_back(box, rank);
+        // linearize vector of vectors
+        std::vector<std::pair<BoundingBox<spacedim>, unsigned int>>
+          boxes_and_ranks;
 
-          // pack boxes into r-tree
-          const auto tree = pack_rtree(boxes_and_ranks);
+        for (unsigned rank = 0; rank < global_bboxes_to_be_used->size(); ++rank)
+          for (const auto &box : (*global_bboxes_to_be_used)[rank])
+            if (is_valid(box))
+              boxes_and_ranks.emplace_back(box, rank);
 
-          // loop over all entities
-          for (unsigned int i = 0; i < entities.size(); ++i)
-            {
-              // create a bounding box with tolerance
-              const auto bb =
-                BoundingBox<spacedim>(entities[i]).create_extended(tolerance);
+        // pack boxes into r-tree
+        const auto tree = pack_rtree(boxes_and_ranks);
 
-              // determine ranks potentially owning point/bounding box
-              std::set<unsigned int> my_ranks;
+        // loop over all entities
+        for (unsigned int i = 0; i < entities.size(); ++i)
+          {
+            // create a bounding box with tolerance
+            const auto bb =
+              BoundingBox<spacedim>(entities[i]).create_extended(tolerance);
+
+            // determine ranks potentially owning point/bounding box
+            std::set<unsigned int> my_ranks;
 
-              for (const auto &box_and_rank :
-                   tree | boost::geometry::index::adaptors::queried(
-                            boost::geometry::index::intersects(bb)))
-                my_ranks.insert(box_and_rank.second);
+            for (const auto &box_and_rank :
+                 tree | boost::geometry::index::adaptors::queried(
+                          boost::geometry::index::intersects(bb)))
+              my_ranks.insert(box_and_rank.second);
+
+            for (const auto rank : my_ranks)
+              ranks_and_indices.emplace_back(rank, i);
+          }
+      };
 
-              for (const auto rank : my_ranks)
-                ranks_and_indices.emplace_back(rank, i);
+#if defined(DEAL_II_WITH_ARBORX)
+      if (global_bboxes.size() == 1)
```

Comment:
If we already have gathered the bounding boxes from all processes, we could still use `ArborX` but that might be something for a follow-up to explore how it compares with the other implementation in terms of performance.

Reply:
That was my first implementation, but I must admit didn't compare the two.

I could set a comparison up by adapting the test https://github.com/dealii/dealii/blob/master/tests/grid/distributed_compute_point_locations_03.cc and compare the numbers for `ArborX::DistributedTree::query()` against what is now on master. Does that sounds reasonable ?